Automattic/jetpack
# Search: add connection error notice

#### Changes proposed in this Pull Request:
* Add connection error notice to the Search plugin

#### Other information:


#### Jetpack product discussion
p9dueE-5zL-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
_Cleaning transients:_
```
delete from wp_options where option_name like '%jetpack_connection_error_reporting_gate_%';
```

1. Use a site with a free plan or one without the Search plan purchased.
2. Connect Jetpack, go to "Broken Token" and set an invalid blog token.
3. Go to "Jetpack -> My Jetpack" to trigger the connection error.
4. Go to "Jetpack -> Search", confirm that the error notice shows up, and it looks good.
5. Click "Restore", the notice should say "Reconnecting Jetpack" instead, and display the spinning loader.
6. The page should reload, the notice should disappear.
7. Remove the transients using the SQL query from above (or start on a new site ¯\\\_(ツ)\_\/¯ ).
8. Go to "Jetpack -> Search", add `&new_pricing_202208=1` to the URL to see the "New Pricing" version of the Search page.
9. Confirm that you see the error notice, and it looks good.
10. Click "Restore", the notice should say "Reconnecting Jetpack" instead, and display the spinning loader.
11. The page should reload, the notice should disappear.
12. Purchase the "Search" plan.
13. Go to "Jetpack Debug -> Connection Errors" and create a fake "verified" error.
14. Go to "Jetpack -> Search" and confirm that the error shows up, and it looks good.
15. Click "Restore", you should be taken to the complete reconnect flow. Confirm you're successfully reconnected.
